lib/jekyll/static_file.rb

```diff
@@ -80,7 +80,7 @@ def write(dest)
 
       FileUtils.mkdir_p(File.dirname(dest_path))
       FileUtils.rm(dest_path) if File.exist?(dest_path)
-      FileUtils.cp(path, dest_path)
+      FileUtils.copy_entry(path, dest_path)
```

Comment:
Could you please wrap this around a `@site.safe` call? We cannot use `copy_entry` in safe mode.

```ruby
if @site.safe || Jekyll.env.start_with?("prod")
  FileUtils.cp(path, dest_path)
else
  FileUtils.copy_entry(path, dest_path)
end
```

The  `Jekyll.env.start_with? "prod"` ensures production builds (where the destination and source may not be on the same host) work as expected.